src/watchtower.ts

```diff
@@ -0,0 +1,76 @@
+import { program } from "@commander-js/extra-typings";
+import { ReplayBlockOptions, ReplayTxOptions, RunOptions } from "./types";
+import { replayBlock, replayTx, run } from "./modes";
+
+async function main() {
+  program
+    .name("watchtower")
+    .description("Monitoring composable cows on the blockchain 🐮")
+    .version("0.2.0");
+
+  program
+    .command("run")
+    .description("Run the watchtower")
+    .requiredOption("--rpc <rpc...>", "Chain RPC endpoints to monitor")
+    .requiredOption(
+      "--contract <contractAddress>",
+      "ComposableCoW contract address"
+    )
+    .requiredOption(
+      "--deployment-block <deploymentBlock...>",
+      "Block number at which the contracts were deployed"
+    )
+    .requiredOption(
+      "--page-size <pageSize>",
+      "Number of events to fetch per page",
+      "5000"
+    )
+    .option("--publish", "Publish orders to the OrderBook API", true)
+    .action((options: RunOptions) => {
+      // Need to assert that the RPCs and deployment blocks are the same length
+      const { rpc, deploymentBlock } = options;
+
+      if (rpc.length !== deploymentBlock.length) {
+        throw new Error("RPC and deployment blocks must be the same length");
+      }
```

Comment:
That requirement is not mentioned in the option descriptions.
Also, now I realized the multiple RPCs and deployment blocks are meant to be 1 per network, but that's not clear from the options.

What about instead the format is changed a bit to make it more clear and avoid a mistake I would certainly make to pass the blocknumber for a different network given that the order has to be followed in two different parameter groups.

Something like: `--chainInfo: 1983712:https://rpc.bla.bla` where the first part is the block number and the second is the rpc endoint.

Reply:
Added help text before the options in relation to the RPC / deployment block orders.

Will consider the `--chainInfo` with future PRs that are likely to allow monolithic (watch all chains), and also "run an instance per chain".